engine/src/main/java/com/datatorrent/stram/engine/Reservoir.java

```diff
@@ -23,20 +23,26 @@
  *
  * @since 0.3.2
  */
-public interface Reservoir
+public interface Reservoir<T>
 {
   /**
    * the count of elements in this SweepableReservoir.
    *
    * @return the count
    */
-  public int size();
+  int size(final boolean dataTupleAware);
```

Comment:
@gauravgopi123 Other than for CircularBufferReservoir this parameter will not be supported. Getting **exact** queue size is way too expensive, so SpscArrayQueue gives an upper estimate on the queue size. Filtering out control tuples is even more costly.